datafusion/expr/src/udaf.rs

```diff
@@ -606,3 +607,49 @@ impl AggregateUDFImpl for AggregateUDFLegacyWrapper {
         (self.accumulator)(acc_args)
     }
 }
+
+pub trait AggregateUDFExprBuilder {
```

Comment:
Also, since it works with `Expr` (not just `AggregateUDF`) I wonder if we should call it something more general like `AggregateExprBuilder` 🤔 

Also, I think we should document this trait with some examples, and specifically documenting the behavior of what happens  when the Expr is *not* an `Expr::AggregateFunction` (the code silently ignores the expr)

Something like this maybe

```rust
/// Trait to assist building aggregate expressions.
///
/// # Error handing
/// (what happens if the Expr is not an aggregate)
///
/// # Example
/// ```
/// show it in use
/// ```
```